### ISSUE TITLE
[Issue]: With v10.9.4, custom folder names in show libraries are not being placed into their own Season entry and being grouped together into one big Season entry with one "season.nfo" file being created for the Show entry.

### ISSUE THREAD

**Issue by maru801** [jellyfin/jellyfin#11916]
### Please describe your bug

I have been using a show library structure that goes against Jellyfin's method of naming each season folder as "Season ##". This was fine as I provided my own metadata to circumvent the obvious that Jellyfin wouldn't be able to work with my naming structure for the folder(s).

Before v10.9.4, Jellyfin would treat each folder as their own Season and display them as such (with broken metadata at times at the time of a scan, as expected) no matter their name. I would then go and replace the metadata and would get a "season.nfo" file in each folder.

With v10.9.4, all folders not named "season ##" will get grouped together. What's more, if I switch the names of all the folders to be in the correct "season ##" format, the season.nfo file in each of them that has metadata will be deleted and replaced with a blank season.nfo causing all my custom metadata to be lost.

To provide context, this is an example of how I had my folders structured for shows:

1. Show-Name
    - custom-season-folder-name-1
            - ep1.mkv
            - ep2.mkv
            - season.nfo
    - custom-season-folder-name-2
            - ep1.mkv
            - ep2.mkv
            - season.nfo
    - custom-season-folder-name-3
            - ep1.mkv
            - ep2.mkv
            - season.nfo

Scanning this structure in v10.9.4 causes the show to group all custom folders into a single season and won't allow for any custom metadata to pull them out into their own season views. After the scan, an extra season.nfo will be created in the top-most show directory with a random season number written in it (same directory as the tvshow.nfo file for the whole show).

I was using the folder names to identify my media in a certain way. Is this no longer supported? Is there a way to change the folder(s) names back to "Season ##" and keep the metadata? I've tried copying them back after naming them correctly, but Jellyfin will not pull the metadata from them, so I have to go back and redo each season for each show that I have.

### Reproduction Steps

1. Rename multiple season folders of a show in a show library to be anything but "Season ##".
2. Go to the show's page in Jellyfin. Run any scan on the entire show.
3. All season not named "Season ##" will be grouped together into a single season entry.

### Jellyfin Version

10.9.0

### if other:

10.9.4

### Environment

```markdown
- OS: Truenas Scale (Debian)
- Linux Kernel: 6.6.29
- Virtualization: Kubernetes
- Clients: Any
- Browser: Any
- FFmpeg Version: FFmpeg version included with v10.9.4 image
- Playback Method: N/A
- Hardware Acceleration: VAAPI
- GPU Model: AMD R7 5750G's iGPU
- Plugins: N/A
- Reverse Proxy: none
- Base URL: none
- Networking: bridge
- Storage: local
```

### Jellyfin logs

```shell
[2024-06-01 23:45:15.902] [INF] [38] Emby.Server.Implementations.Library.LibraryManager: Removing item, Type: "Season", Name: "<redacted>", Path: "/Shows/<redacted> [TV+BD]/1. <redacted> [BD]", Id: 65547019-7452-3d00-3d9a-9289aee89289
[2024-06-01 23:45:15.944] [INF] [38] Emby.Server.Implementations.Library.LibraryManager: Removing item, Type: "Season", Name: "<redacted>", Path: "/Shows/<redacted> [TV+BD]/3. <redacted> [BD]", Id: 7797f669-1a24-de46-9228-4eba5d2b769e
[2024-06-01 23:45:15.948] [INF] [38] Emby.Server.Implementations.Library.LibraryManager: Removing item, Type: "Season", Name: "<redacted>", Path: "/Shows/<redacted> [TV+BD]/4. <redacted> [BD]", Id: dd95dd0c-38e8-bb34-be40-ad058144b3f1
[2024-06-01 23:45:15.982] [INF] [38] Emby.Server.Implementations.Library.LibraryManager: Removing item, Type: "Season", Name: "<redacted>", Path: "/Shows/<redacted> [TV+BD]/2. <redacted> [TV]", Id: 6559c644-d1e9-6b7e-d698-a317f91c22b1
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Shadowghost** [comment]
Likely caused by #11719 but I tend to say this is a non isue since your file structure is somewhat unsupported.
Looking at your season folder name and my local testing, #11748 should fix it but that will not come to 10.9.z

**Comment by maru801** [comment]
Yeah, I figured. The fix you point to seems promising for my use case however.

I'll revert back to v10.9.3 for now and wait for it to get implemented in the future.

If it comes down to it and I need to update my server for whatever reason, I'll bite the bullet and just redo my metadata bit by bit.
I really wish there was a library type that didn't pull metadata (unless manually requested and identified by the user) and just displayed the folder structure, but shown on screen as the "shows" library type. I've tried other library types for this case, but the episodes get displayed in a wonky way that's not appealing to browse at all.

**Comment by jhparker88** [comment]
> Likely caused by https://github.com/jellyfin/jellyfin/pull/11719 but I tend to say this is a not really an issue since your file structure is unsupported.

I have a similar series structure especially for long running Japanese shows like (浅見光彦シリーズ or 赤い霊柩車)  which normally run over several years and don't adhere the rigid label of a "season" (more precise a "season" or "temporada" or "staffel" are not known to these shows).

**Comment by Shadowghost** [comment]
see the linked pr above. this should fix it.

**Comment by SteveDinn** [comment]
Somebody on the forums mentioned that this might be the root cause of my issue that I described here: https://forum.jellyfin.org/t-movies-have-disappeared-from-jellyfin-but-files-still-exist

It doesn't *sound* like the same issue to me, but perhaps someone with a deeper understanding of the issue might know better.

TL;DR: Files on disk haven't changed and are still accessible by the jellyfin container, but I see this message in my logs, and the media have disappeared from Jellyfin.
```
[WRN] [279] MediaBrowser.Controller.Entities.BaseItem: Unable to find linked item at path [path-to-media-file]
```

**Comment by BrianCrowe3** [comment]
I've noticed some of my shows randomly have a season.nfo file in the root directory now, even though I'm using "Season 01, etc" naming scheme. It's not really causing any noticeable issues, but it is very weird.

If it's related to trying to force everything into a season folder, it definitely feels like a usability regression because sometimes you want to include something with a show, but it doesn't fit into any of the seasons, and hiding it in the extras folder so no one ever sees it is a less than ideal solution.

![explorer_5t8Nbbif7T](https://github.com/jellyfin/jellyfin/assets/115080087/543572be-e42c-4ba8-86ef-cb1c15dd6645)

**Comment by maru801** [comment]
From that screenshot, I'm going to guess that the "April Fool's FanSub" folder you have is the culprit. I haven't tested if a season folder named "Extras" will also cause this. I get the same file in the same location as described in the first post of this issue. This should be fixed with the pull request @Shadowghost mentioned earlier.

**Comment by BrianCrowe3** [comment]
> I haven't tested if a season folder named "Extras" will also cause this.
What the extras folder does is just create a "Special Features" section on the show page below Cast & Crew. At this point, 10.9.x has caused me so much grief that I'm just done updating jellyfin at all unless there's a significant security or client compatibility issue. I cant keep having my nfo files get fragged every other update, even with backups it's extremely annoying.

**Comment by maru801** [comment]
This issue has been fixed by the release of v10.9.5.
